op-chain-ops/genesis/layer_one.go

```diff
@@ -56,7 +56,16 @@ func BuildL1DeveloperGenesis(config *DeployConfig, dump *gstate.Dump, l1Deployme
 	SetPrecompileBalances(memDB)
 
 	if dump != nil {
-		for address, account := range dump.Accounts {
+		for addrstr, account := range dump.Accounts {
+			if !common.IsHexAddress(addrstr) {
+				// TODO(Seb) confirm this is what we want here
+				//   Changes in https://github.com/ethereum/go-ethereum/pull/28504
+				//   now also add accounts to the Dump with "pre(<AddressHash>)" as key
+				//   if the address itself is nil.
+				//   So this might be a pre-image key, in which case we skip.
+				continue
+			}
+			address := common.HexToAddress(addrstr)
```

Comment:
We currently generate the l1-allocs with the `bedrock-devnet` script, which in turn calls a [forge script `runWithStateDump()`](https://github.com/ethereum-optimism/optimism/blob/dcdf044f3a5ce39afdf67e9efbb8ec3f1b9b0be4/bedrock-devnet/devnet/__init__.py#L134) to generate the dump. This one probably doesn't generate any `pre(...)` entries. However, because someone _could_ in theory call `BuildL1DeveloperGenesis` with a `Dump` that was created with a `StateDB.Dump` call, without the `OnlyWithAddresses` config flag, I will leave this skip in here as a precaution.

Will remove the TODO but leave comment open for reviewers to resolve.